### ISSUE TITLE
Containment Bug?

### ISSUE THREAD

**Issue by sbryfcz** [tmpvar/2d-polygon-boolean#12]
I am having problems with clipping when a subject is completely contained within the clipping polygon or vice versa. It seems to return just an empty array. This doesn't seem like expected behavior. I'd expect the result of the union to be the polygon that was contained. Thoughts? Thanks for the great library.

**Comment by tmpvar** [comment]
can you please provide a test case so I don't misspeak when offering advice / determining if this is a bug?

**Comment by sbryfcz** [comment]
Sorry. I worked one up but forgot to include it! You can swap clip with subject and its still the same. This should be one square completely inside another square.

```javascript
var clip = Polygon([
  Vec2(0, 0),
  Vec2(100, 0),
  Vec2(100, 100),
  Vec2(0, 100),
  Vec2(0, 0)
]);

 var subject = Polygon([
  Vec2(10, 10),
  Vec2(20, 10),
  Vec2(20, 20),
  Vec2(10, 20),
  Vec2(10, 10)
]);

// union is an array of Polygons 
var union = subject.clip(clip, 'union');
```

**Comment by tmpvar** [comment]
Yeah, this is totally a bug!

**Comment by sbryfcz** [comment]
Glad I'm not crazy ;)

Not sure how difficult it is to fix. Looks like there is another npm package that could possibly help: https://www.npmjs.com/package/2d-polygon-contains-polygon

**Comment by tmpvar** [comment]
So the solution to this problem should be pretty easy, but do be aware that there are other degeneracies with this module. see #10 (I'd be happy to get some help!)

Yeah, that module would be useful :) the main question is do we want to provide special windings to these polygons? i.e. in the case of a boolean not (cut) should we wind the inner polygon to be the reverse of the outer? (this can be easily accomplished by testing the sign returned from https://www.npmjs.com/package/2d-polygon-area)

* `union` - discard contained polygon, return container
* `and` - discard container polygon, return contained
* `not` - return both

**Comment by tmpvar** [comment]
actually we don't need `2d-polygon-contains-polygon`, we should be able to do a simple test if the result is `[]`.

should go something like:

```
if empty result set and empty isects
  if clipContains(subject[0])
    ...
  else if subjectContains(clip[0])
    ...
```

**Comment by sbryfcz** [comment]
Yeah. I guess I wassn't sure how difficult the clipContains and subjectContains functions would be to come up with.

**Comment by tmpvar** [comment]
they should be ready to go https://github.com/tmpvar/2d-polygon-boolean/blob/master/2d-polygon-boolean.js#L211-L212

If you want to chat in realtime I've made a gitter room: https://gitter.im/tmpvar/2d

**Comment by sbryfcz** [comment]
Sorry cant chat at the moment. 

Alright so I've been playing with 2d-polygon-boolean instead of polygon.clip (which I guess is older?). It seems like winding now matters a lot. What is the guidance on that? The polygon.clip library worked (for my example anyways) but now I'm getting weird results. Almost like the 'and' operation is an 'or' operation. I changed the winding and things are better but I could use some guidance. I dont see it in your wiki.

**Comment by tmpvar** [comment]
if you want something that works, right at this moment, you should look up clipper.js

**Comment by tmpvar** [comment]
published version 1.0.1 which includes fixes to the containment issues raised in this issue.
